library/ix-dev/community/rsyncd/templates/_validation.tpl

```diff
@@ -0,0 +1,83 @@
+{{- define "rsync.validation" -}}
+  {{- $reservedParams := (list
+                          "port" "use chroot" "pid file"
+                          "max connections"
+                          ) -}}
+
+  {{- if .Values.rsyncConfig.logToStdout -}}
+    {{- $reservedParams = mustAppend $reservedParams "log file" -}}
+  {{- end -}}
+
+  {{- range .Values.rsyncConfig.auxParams -}}
+    {{- include "rsync.aux.validation" (dict "aux" .) -}}
+
+    {{- if mustHas .param $reservedParams -}}
+      {{- fail (printf "Rsync - Overriding parameter [%v] is not allowed." .param) -}}
+    {{- end -}}
+  {{- end -}}
+
+  {{- if not .Values.rsyncModules -}}
+    {{- fail "Rsync - At least one module must be configured." -}}
+  {{- end -}}
+{{- end -}}
+
+{{- define "rsync.module.validation" -}}
+  {{- $mod := .mod -}}
+  {{- if not $mod.name -}}
+    {{- fail "Rsync - [Module Name] is required." -}}
+  {{- end -}}
+
+  {{- if not (mustRegexMatch "^[a-zA-Z0-9]+([_-]*[a-zA-Z0-9]+)+$" $mod.name) -}}
```

Comment:
We should have this in questions schema too please

Reply:
Added